cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -133,6 +133,13 @@ set_property(TARGET compiler-cpp PROPERTY dialect_cpp2b "-std=c++2b"
 # Flag for disabling strict aliasing rule in C and C++
 set_compiler_property(PROPERTY no_strict_aliasing -fno-strict-aliasing)
 
+# Extra warning options
+if (FORCE_WARNING_AS_ERRORS)
+  set_property(TARGET compiler PROPERTY cc_extra_warning_options "-Werror")
+  set_property(TARGET asm PROPERTY as_extra_warning_options -Werror -Wa,--fatal-warnings)
+endif()
```

Comment:
Let's set property without a check for `FORCE_WARNING_AS_ERRORS` - and only check for `FORCE_WARNING_AS_ERRORS` when we access this property in (in `zephyr/CMakeLists.txt`)